### PR TITLE
Send payment update message after creating case from exception record

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/BPS-824

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java [hunk 1]
```diff
@@ -488,6 +506,150 @@ void should_report_errors_when_form_type_is_null() {
...
+        data.put("poBox", "12345");
+        data.put("journeyClassification", EXCEPTION.name());
+        data.put("formType", "A1");
+        data.put("deliveryDate", "2019-09-06T15:30:03.000Z");
+        data.put("openingDate", "2019-09-06T15:30:04.000Z");
+        data.put("scannedDocuments", TestCaseBuilder.document("https://url", "name"));
+        data.put("scanOCRData", TestCaseBuilder.ocrDataEntry("key", "value"));
+        data.put(ExceptionRecordFields.CONTAINS_PAYMENTS, YesNoFieldValues.NO); // no payments!
+        data.put(ExceptionRecordFields.ENVELOPE_ID, envelopeId);
+        data.put(ExceptionRecordFields.PO_BOX_JURISDICTION,jurisdiction);
```
Comment (aliveni) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java]: building case data can be extracted and used in 2 tests

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java [hunk 2]
```diff
@@ -207,6 +220,41 @@ private ProcessResult createNewCase(
         }
     }
 
+    private void handlePayments(CaseDetails exceptionRecord, long newCaseId) {
+
+        boolean containsPayments =
+            Objects.equals(
+                exceptionRecord.getData().get(ExceptionRecordFields.CONTAINS_PAYMENTS).toString(),
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: Can we extract `exceptionRecord.getData()` into a variable ? It’s repeated thrice

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java (hunk 1 above)
Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java]: This is bound to happen some time for sure. It's all over the place. I've started in separate branch but never had a chance to progress further as more changes coming in

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java [hunk 3]
```diff
@@ -80,14 +87,17 @@ public ProcessResult process(CcdCallbackRequest request, String idamToken, Strin
...
-            .getValidation(request.getCaseDetails())
+            .getValidation(exceptionRecordData)
             .map(exceptionRecord -> createNewCase(
                 exceptionRecord,
                 serviceConfigItem,
                 request.isIgnoreWarnings(),
                 idamToken,
-                userId
+                userId,
+                exceptionRecordData
```
[the remark this answers is not in the document]
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: sadly `exceptionRecord` is the request for transformation endpoint, not entire exception record we get from CCD

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java [hunk 4]
```diff
@@ -238,15 +238,14 @@ private void handlePayments(CaseDetails exceptionRecord, long newCaseId) {
...
-            paymentsPublisher
-                .send(
-                    new UpdatePaymentsCommand(
-                        Long.toString(exceptionRecord.getId()),
-                        Long.toString(newCaseId),
-                        envelopeId,
-                        jurisdiction
-                    )
-                );
+            paymentsPublisher.send(
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: it did raise me an eyebrow before 🙃

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java [hunk 5]
```diff
@@ -48,19 +52,22 @@
...
     private final AuthTokenGenerator s2sTokenGenerator;
+    private final PaymentsPublisher paymentsPublisher;
     private final CoreCaseDataApi ccdApi;
 
     public CreateCaseCallbackService(
         CreateCaseValidator validator,
         ServiceConfigProvider serviceConfigProvider,
         TransformationClient transformationClient,
         AuthTokenGenerator s2sTokenGenerator,
+        PaymentsPublisher paymentsPublisher,
```
[the remark this answers is not in the document]
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: didn't realise we were ordering class dependencies by usage order ;)

[the remark this answers is not in the document]
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: we're not. just thought i'm gonna be picky and guess the reason

**Comment by rkondratowicz** [comment]
closing and opening a new one - helm issues.

**Pull request closed without merging**